src/main/java/org/broadinstitute/hellbender/tools/spark/pathseq/PSBuildReferenceTaxonomyUtils.java

```diff
@@ -212,20 +213,34 @@ private static void updateAccessionTaxonProperties(final String accession, final
         } else {
             taxonProperties = new PSPathogenReferenceTaxonProperties();
         }
-        taxonProperties.accessions.add(name);
-        taxonProperties.length += length;
+        taxonProperties.addAccession(name, length);
         taxIdToProperties.put(accession, taxonProperties);
     }
 
     /**
+     * Removes nodes not in the tree from the tax_id-to-properties map
+     */
+    static void removeUnusedTaxIds(final Map<String, PSPathogenReferenceTaxonProperties> taxIdToProperties,
+                                   final PSTree tree) {
+        final Set<String> unusedNodes = new HashSet<>(taxIdToProperties.keySet());
+        unusedNodes.removeAll(tree.getNodeIDs());
+        taxIdToProperties.keySet().removeAll(unusedNodes);
```

Comment:
Isn't this the same as `taxIdToProperties.keySet().retainAll(tree.getNodeIDs());`?